src/bin/poll_oneoff.rs

```diff
@@ -100,6 +100,59 @@ unsafe fn test_stdin_read() {
     );
 }
 
+unsafe fn test_stdout_stderr_write() {
+    let clock = wasi_unstable::raw::__wasi_subscription_u_clock_t {
+        identifier: CLOCK_ID,
+        clock_id: wasi_unstable::CLOCK_MONOTONIC,
+        timeout: 5_000_000u64, // 5 milliseconds
+        precision: 0,
+        flags: 0,
+    };
+    let stdout_readwrite = wasi_unstable::raw::__wasi_subscription_u_fd_readwrite_t {
+        fd: wasi_unstable::STDOUT_FD,
+    };
+    let stderr_readwrite = wasi_unstable::raw::__wasi_subscription_u_fd_readwrite_t {
+        fd: wasi_unstable::STDERR_FD,
+    };
+    let in_ = [
+        wasi_unstable::Subscription {
+            userdata: CLOCK_ID,
+            type_: wasi_unstable::EVENTTYPE_CLOCK,
+            u: wasi_unstable::raw::__wasi_subscription_u { clock },
+        },
+        wasi_unstable::Subscription {
+            userdata: 1,
+            type_: wasi_unstable::EVENTTYPE_FD_READ,
+            u: wasi_unstable::raw::__wasi_subscription_u {
+                fd_readwrite: stdout_readwrite,
+            },
+        },
+        wasi_unstable::Subscription {
+            userdata: 2,
+            type_: wasi_unstable::EVENTTYPE_FD_READ,
+            u: wasi_unstable::raw::__wasi_subscription_u {
+                fd_readwrite: stderr_readwrite,
+            },
+        },
+    ];
+    let out = poll_oneoff_impl(&in_, 1);
```

Comment:
stdout and stderr should be writable, so as far as I understand, poll should return immediately with both fds. Did you run the tests?

Reply:
Yep, that's correct. You were probably looking at the initial draft commit which included by mistake an `EVENTTYPE_FD_READ` on `stdout/err` instead of an `EVENTTYPE_FD_WRITE`. This has now been fixed.